palette/src/rgb/packed.rs

```diff
@@ -0,0 +1,305 @@
+/// let float = 0.5f32 * 255.0;
+/// assert_eq!(0x7F, float as u32);
+/// assert_eq!(0x80, float.round() as u32);
+/// ```
+///
+/// When an `Rgb` type is packed, the alpha value will be `0xFF` in the
+/// corresponding `u32`. Converting from a packed color type back to an `Rgb`
+/// type will disregard the alpha value.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Pixel)]
+#[palette_internal]
+#[repr(C)]
+pub struct Packed<C: RgbChannels = channels::Argb> {
+    /// The sRGB color packed into a `u32`.
+    pub color: u32,
+
+    /// The channel ordering for red, green, blue, and alpha components in the
+    /// packed integer; can be `Abgr`, `Argb`, `Bgra`, or `Rgba`. See
+    /// [RgbChannels](trait.RgbChannels.html).
+    #[palette_unsafe_zero_sized]
+    pub channel_order: PhantomData<C>,
+}
+
+/// Splits and combines RGB(A) types with some channel ordering. Channels may be
+/// ordered as `Abgr`, `Argb`, `Bgra`, or `Rgba`.
+pub trait RgbChannels {
+    /// Split RGBA components into a `(u8, u8, u8, u8)` tuple.
+    fn split_rgb<S: RgbStandard>(rgb: Rgba<S, u8>) -> (u8, u8, u8, u8);
+    /// Create an RGBA color from a `(u8, u8, u8, u8)` tuple.
+    fn combine_rgb<S: RgbStandard>(channels: (u8, u8, u8, u8)) -> Rgba<S, u8>;
+}
+
+impl<C: RgbChannels> From<u32> for Packed<C> {
+    fn from(color: u32) -> Self {
+        Packed {
+            color,
+            channel_order: PhantomData,
+        }
+    }
+}
+
+impl<C: RgbChannels> Into<u32> for Packed<C> {
+    fn into(self) -> u32 {
+        self.color
+    }
+}
+
+impl<S, C> Into<Rgb<S, u8>> for Packed<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn into(self) -> Rgb<S, u8> {
+        let bytes: [u8; 4] = self.color.to_be_bytes();
+        let color: Alpha<Rgb<S, u8>, u8> = C::combine_rgb((bytes[0], bytes[1], bytes[2], bytes[3]));
+        Rgb::from_components((color.red, color.green, color.blue))
+    }
+}
+
+impl<S, C> From<Rgb<S, u8>> for Packed<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn from(color: Rgb<S, u8>) -> Self {
+        // Alpha value of 0xFF declared here as specified in doc for `Packed`
+        let src: Alpha<Rgb<S, u8>, u8> =
+            Rgba::from_components((color.red, color.green, color.blue, 0xFF));
+        Self::from(src)
+    }
+}
+
+impl<S, C> Into<Rgba<S, u8>> for Packed<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn into(self) -> Rgba<S, u8> {
+        let bytes: [u8; 4] = self.color.to_be_bytes();
+        C::combine_rgb((bytes[0], bytes[1], bytes[2], bytes[3]))
+    }
+}
+
+impl<S, C> From<Rgba<S, u8>> for Packed<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn from(color: Rgba<S, u8>) -> Self {
+        let bytes = C::split_rgb(color);
+        Packed {
+            color: 0x0000_0000
+                | ((bytes.0 as u32) << 24)
+                | ((bytes.1 as u32) << 16)
+                | ((bytes.2 as u32) << 8)
+                | (bytes.3 as u32),
```

Comment:
I saw you switched to `to_be_bytes` above. I assume this can be replaced with `from_be_bytes`, then. :slightly_smiling_face:

Reply:
Good catch 👍